app/models/miq_provision.rb

```diff
@@ -31,13 +31,27 @@ class MiqProvision < MiqProvisionTask
 
   CLONE_SYNCHRONOUS     = false
   CLONE_TIME_LIMIT      = 4.hours
-  SUPPORTED_EMS_CLASSES = %w( ManageIQ::Providers::Vmware::InfraManager
-                              ManageIQ::Providers::Redhat::InfraManager
-                              ManageIQ::Providers::Amazon::CloudManager
-                              ManageIQ::Providers::Openstack::CloudManager
-                              ManageIQ::Providers::Microsoft::InfraManager
-                              ManageIQ::Providers::Google::CloudManager
-                              ManageIQ::Providers::Azure::CloudManager)
+
+  def self.supported_ems_classes
+    @supported_ems_classes ||= %w( ManageIQ::Providers::Vmware::InfraManager
```

Comment:
@Fryguy I'm not sure about modifying class variables. I remember that being not thread safe. Is that still the case?
Then I would rather use `Thread.current[]` for storing that....